test/index.test.js

```diff
@@ -0,0 +1,150 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const dcPlugin = require('../lib/index')
+const dc = require('diagnostics_channel')
+const sget = require('simple-get').concat
+
+test('Should call publish when route is registered', t => {
+  t.plan(6)
+  const fastify = Fastify()
+
+  const onRouteChannel = dc.channel('http.fastify.onRoute')
+  let timesCalled = 0
+  const routeOptions = [
+    {
+      method: 'GET',
+      url: '/test/1',
+      path: '/test/1',
+      routePath: '/1',
+      prefix: '/test',
+      logLevel: '',
+      attachValidation: false
+    },
+    {
+      method: 'POST',
+      url: '/test/2',
+      path: '/test/2',
+      routePath: '/2',
+      prefix: '/test',
+      logLevel: '',
+      attachValidation: false
+    }
```

Comment:
Might be good to verify you get the routing fragments in here where relevant, rather than just absolute values.